### PR TITLE
Configuration setup

### PR DESCRIPTION
**Pull request by Kney-Delach**
Completed configuration setup with lua, some errors may arise due to not pulling the project recursively.

### CODE REVIEW

File: Hidden-Engine/Configuration/Premake/Sandbox.lua [hunk 1]
```diff
@@ -0,0 +1,68 @@
+project "Sandbox"
+	location "../../Sandbox"
```
[the remark this answers is not in the document]
Comment (Kney-Delach) [reply Hidden-Engine/Configuration/Premake/Sandbox.lua]: Fixed error, was related to backend engine networking protocols.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	url = https://github.com/Kney-Delach/Glad
+[submodule "Hidden-Engine/Hidden-Engine/Dependencies/spdlog"]
+	path = Hidden-Engine/Hidden-Engine/Dependencies/spdlog
+	url = https://github.com/gabime/spdlog
+[submodule "Hidden-Engine/Hidden-Engine/Dependencies/GLFW"]
+	path = Hidden-Engine/Hidden-Engine/Dependencies/GLFW
+	url = https://github.com/Kney-Delach/glfw
+[submodule "Hidden-Engine/Hidden-Engine/Dependencies/GLM"]
+	path = Hidden-Engine/Hidden-Engine/Dependencies/GLM
+	url = https://github.com/g-truc/glm
```
Comment (Arangear) [inline .gitmodules]: I can't believe you've done this. It's both amazing and terrifying.

File: Hidden-Engine/Hidden-Engine/Src/Hidden/Core/Application.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Created: [06 / 02 / 2020] 
+// Updated: [06 / 02 / 2020]
```
Comment (Arangear) [inline Hidden-Engine/Hidden-Engine/Src/Hidden/Core/Application.cpp]: Lies.

File: Hidden-Engine/Premake5.lua [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	architecture "x64"
+
+	configurations 
+	{
+		"Windows-Debug",
+		"PS4-Debug",
+		"Windows-Release",
+		"PS4-Release"
+	}
+
```
Comment (Arangear) [inline Hidden-Engine/Premake5.lua]: An important contextual comment.

File: LICENSE [hunk 5]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 Hidden-Lab
+Copyright (c) 2020 Hidden Lab
```
Comment (Arangear) [inline LICENSE]: Wat?

File: Hidden-Engine/Configuration/Premake/Sandbox.lua [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	filter "configurations:Windows-Debug"
+		defines 
+		{
+			"HE_DEBUG",
+			"HE_PLATFORM_WINDOWS"
+		}
+		runtime "Debug"
+		symbols "On"
+
+	filter "configurations:Windows-Release"
```
Comment (llapuras) [inline Hidden-Engine/Configuration/Premake/Sandbox.lua]: dummy comment ......

**Comment by Kney-Delach** [comment]
Decided not to merge this pull request, as configuration files need updating.

**Pull request closed without merging**